Source/Scene/Primitive.js

```diff
@@ -1217,6 +1218,10 @@ define([
             modelMatrix = this.modelMatrix;
         }
 
+        if (!Matrix4.equals(this.modelMatrix, Matrix4.IDENTITY) && ((this._numberOfInstances > 1 && !this._validModelMatrix) || (frameState.mode === SceneMode.SCENE2D || frameState.mode === SceneMode.COLUMBUS_VIEW))) {
+            throw new DeveloperError('If the model matrix is changed after creation, it only affects primitives with one instance and only in 3D mode.');
```

Comment:
I would change this error (and the doc for it) to `Primitive.modelMatrix is only supported in 3D mode and only for single geometry instances`.